pkg/lockfile/parse-pnpm-lock.go

```diff
@@ -77,6 +81,17 @@ func extractPnpmPackageNameAndVersion(dependencyPath string) (string, string) {
 	return name, version
 }
 
+func parseNameAtVersion(value string) (name string, version string) {
+	// look for pattern "name@version", where name is allowed to contain zero or more "@"
+	matches := regexp.MustCompile("^(.+)@([^@/]+)$").FindStringSubmatch(value)
```

Comment:
Can we be a bit stricter here to avoid potentially matching on too many things? Some things we can do:

- Require colon at the end of this. i.e. ":$" 
- Match on `[\d\.]` instead of `[^@/]` for the version bits. 
- Anything else?

Reply:
Sounds like a good idea, I will see what I can do to make this match more specific.

Reply:
The `name` parameter does not include the `:` at the end in this case as its already been removed so I can't add that match however.